src/modules/users/delete-user/delete-user.js

```diff
@@ -0,0 +1,86 @@
+(function() {
+    "use strict";
+
+    angular
+        .module("TendrlModule")
+        .controller("deleteUserController", deleteUserController);
+
+    /*@ngInject*/
+    function deleteUserController($rootScope, $scope, $state, userStore, selectedUser, Notifications) {
+
+        var vm = this,
+            userId;
+
+        vm.cancelModal = cancelModal;
+        vm.closeModal = closeModal;
+        vm.confirmModal = confirmModal;
+        vm.userId = selectedUser.username;
+
+        vm.modalHeader = {
+            "title": "Delete User",
+            "close": vm.closeModal
+        };
+
+        vm.modalFooter = [{
+            "name": "Cancel",
+            "type": "button",
+            "classname": "btn-default",
+            "onCall": vm.cancelModal
+        }, {
+            "name": "Yes",
+            "type": "submit",
+            "classname": "btn-primary",
+            "onCall": vm.confirmModal
+        }];
+
+        /**
+         * @name cancelModal
+         * @desc cancels the modal
+         * @memberOf deleteUserController                
+
+         */
+
+        function cancelModal() {
+            $state.go("users");
+            $rootScope.$emit("modal.done", "cancel");
+        }
+
+        /**
+         * @name closeModal
+         * @desc close the modal
+         * @memberOf deleteUserController                
+
+         */
+        function closeModal() {
+            $rootScope.$emit("modal.done", "close");
+        }
+
+        /**
+         * @name next
+         * @desc takes to next step
+         * @memberOf deleteUserController                
+
+         */
+        function confirmModal() {
+            userStore.deleteUser(selectedUser.username)
+                .then(function(data) {
+                    vm.closeModal();
+                    userStore.getUserList()
+                        .then(function(data) {
+                            if (data !== null) {
+                                selectedUser.userList = data;
```

Comment:
@a2batic Again here selectedUser.userList is not required. Directly broadcast data.